### PR TITLE
Add build helper scripts for PETSc and CGNS

### PR DESCRIPTION
**Pull request by A-CGray**
## Purpose

Adds two bash scripts that automate the configuration and building of PETSc and CGNS. I use these on my local machine and on clusters so that I can ensure a repeatable build.

## Type of change

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (non-backwards-compatible fix or feature)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [x] Documentation update
- [ ] Maintenance update
- [ ] Other (please describe)

## Checklist

- [ ] I have run `flake8` and `black` to make sure the Python code adheres to PEP-8 and is consistently formatted
- [ ] I have formatted the Fortran code with `fprettify` or C/C++ code with `clang-format` as applicable
- [ ] I have run unit and regression tests which pass locally with my changes
- [ ] I have added new tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation

### CODE REVIEW

**Review by marcomangano — COMMENTED** [review]
This looks pretty useful! A first comment on my side is that you could take into some CLI arguments instead of editing the file. You can take a look at the `docker` repo for the logic we use there

**Comment by A-CGray** [comment]
> This looks pretty useful! A first comment on my side is that you could take into some CLI arguments instead of editing the file. You can take a look at the `docker` repo for the logic we use there

Right, I based these scripts off the steps in the docker repo. But the point of these scripts (at least for me) is so that I know I have all the configuration options and build steps written down somewhere. If I made them CLI arguments then I'd be back to having to remember what inputs I used last time I built PETSc

File: installInstructions/buildCGNS.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
+#! /bin/sh
+version= # TODO: PUT THE VERSION OF PETSC YOU WANT TO BUILD HERE (e.g 4.3.0)
+
+# If the user doesn't have an OPT_FLAGS environment variable, define the optimization flags here
+if [ -z ${var+x} ]; then
```
Comment (sseraj) [inline installInstructions/buildCGNS.sh:5]: Shouldn't this just be `-z $OPT_FLAGS`? Same for the other script

File: installInstructions/buildPETSc.sh [hunk 2]
```diff
@@ -0,0 +1,58 @@
+#! /bin/bash
+version= # TODO: PUT THE VERSION OF PETSC YOU WANT TO BUILD HERE (e.g 3.19.5)
```
Comment (sseraj) [inline installInstructions/buildPETSc.sh:2]: Maybe Marco is referring to adding CLI options for the versions or combinations of opt/debug, real/complex? I'm fine with not adding CLI if the user has to edit the configuration in the file anyway.

File: installInstructions/buildCGNS.sh (hunk 1 above)
Comment (A-CGray) [reply installInstructions/buildCGNS.sh]: Wow yeah the logic here was totally messed up, fixed now using [this stack overflow answer](https://stackoverflow.com/questions/3601515/how-to-check-if-a-variable-is-set-in-bash/17538964#17538964)

File: installInstructions/buildPETSc.sh (hunk 2 above)
Comment (marcomangano) [reply installInstructions/buildPETSc.sh:2]: Yeah exactly, if the file needs to be tweaked anyway then this is fine @A-CGray

**Review by sseraj — DISMISSED** [review]

**Comment by A-CGray** [comment]
You happy with this @marcomangano ?

**Review by marcomangano — APPROVED** [review]
Haven't tried these directly but this is going to be helpful. No need to add more complexity with CLI inputs

**Comment by A-CGray** [comment]
@sseraj

**Review by sseraj — APPROVED** [review]

**Pull request merged**